.github/workflows/build.yml

```diff
@@ -55,47 +51,68 @@ jobs:
-        gzip --best $HLS
-        echo ::set-output name=hls_binary::$HLS.gz
+        HLS_BUILD=$(find dist-newstyle \( -name 'hls' -o -name 'hls.exe' \) -type f)
+        HLS=haskell-language-server-$GHC_VER
+        mv $HLS_BUILD $HLS
+        if [[ "$OSTYPE" == "msys" ]]; then
+          7z a $HLS.zip $HLS
+          echo ::set-output name=path::$HLS.zip
+          echo ::set-output name=content_type::application/zip
+          echo ::set-output name=extension::zip
+        else
+          gzip --best $HLS
+          echo ::set-output name=path::$HLS.gz
+          echo ::set-output name=content_type::application/gzip
+          echo ::set-output name=extension::gz
+        fi
 
     - name: Upload Server Binary
       uses: actions/upload-release-asset@v1.0.2
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         upload_url: ${{ github.event.release.upload_url }}
-        asset_path: ${{ steps.find_server_binary.outputs.hls_binary }}
-        asset_name: haskell-language-server-${{ runner.OS }}-${{ matrix.ghc }}${{env.EXE_EXT}}.gz
-        asset_content_type: application/gzip
+        asset_path: ${{ steps.compress_server_binary.outputs.path }}
+        asset_name: haskell-language-server-${{ runner.OS }}-${{ matrix.ghc }}${{env.EXE_EXT}}.${{ steps.compress_server_binary.outputs.extension }}
+        asset_content_type: ${{ steps.compress_server_binary.outputs.content_type }}
 
     - uses: actions/upload-artifact@v2
       with:
         name: haskell-language-server-${{ runner.OS }}-${{ matrix.ghc }}${{env.EXE_EXT}}.gz
-        path: ${{ steps.find_server_binary.outputs.hls_binary }}
+        path: ${{ steps.compress_server_binary.outputs.path }}
 
     - name: Build Wrapper
       if: matrix.ghc == '8.10.1'
       run: cabal build exe:hls-wrapper -O2 --disable-documentation $WIN_CABAL_ARGS $LINUX_CABAL_ARGS
 
-    - name: Find Wrapper Binary
+    - name: Compress Wrapper Binary
       if: matrix.ghc == '8.10.1'
-      id: find_wrapper_binary
-      shell: bash
+      id: compress_wrapper_binary
       run: |
-        HLS_WRAPPER=$(find dist-newstyle \( -name 'hls-wrapper' -o -name 'hls-wrapper.exe' \) -type f)
-        gzip --best $HLS_WRAPPER
-        echo ::set-output name=hls_wrapper_binary::$HLS_WRAPPER.gz
+        HLS_WRAPPER_BUILD=$(find dist-newstyle \( -name 'hls-wrapper' -o -name 'hls-wrapper.exe' \) -type f)
+        HLS_WRAPPER=haskell-language-server-wrapper
+        mv $HLS_WRAPPER_BUILD $HLS_WRAPPER
+        if [[ "$OSTYPE" == "msys" ]]; then
+          7z a $HLS_WRAPPER.zip $HLS_WRAPPER
+          echo ::set-output name=path::$HLS_WRAPPER.zip
+          echo ::set-output name=content_type::application/zip
+          echo ::set-output name=extension::zip
+        else
+          gzip --best $HLS_WRAPPER
+          echo ::set-output name=path::$HLS_WRAPPER.gz
+          echo ::set-output name=content_type::application/gzip
+          echo ::set-output name=extension::gz
+        fi
```

Comment:
Is it possible to move these into OS-specific step? I mean, is it possible to set and use a binary as an env var? If so, I think it would be better to extract commons... (`.zip` extension, content type, ...)

Reply:
Yeah it's a bit unwieldily, what do you mean by set and use binary as env var though? I avoided splitting them out into OS-specific steps since skipped steps still show up in the UI and so you end up with two "compress binary" steps

Reply:
Oh, I tried it, and noticed that it doesn't work... never mind.